### PR TITLE
[BugFix] Check overflow of BinaryColumn for window function

### LINKED ISSUE #25359: [Bug] Report memory limit when processing a large unbounded window
### Steps to reproduce the behavior (Required)

tpch-100g

```sql
SELECT l_comment, MAX(l_comment) OVER() AS max_comment
FROM lineitem
ORDER BY l_orderkey DESC LIMIT 17985060, 10;
```

### Expected behavior (Required)

```
(1064, 'Internal error: BinaryColumn overflow, size: 4294968248')
```

### Real behavior (Required)

```
(1064, 'Memory of process exceed limit. try consume:21474836480 Used: 37005315480, Limit: 53711618088. Mem usage has exceed the limit of BE')
```

And maybe it will lead to deadly consequence like be crash if there is a projection over the `max_comment` column

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

```
342e9f5ceb
```

### PR DESCRIPTION
**Pull request by liuyehcf**
Fixes #25359

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4

### CODE REVIEW

**Review by silverbullet233 — DISMISSED** [review]

File: be/src/column/binary_column.h [hunk 1]
```diff
@@ -335,6 +343,14 @@ class BinaryColumnBase final : public ColumnFactory<Column, BinaryColumnBase<T>>
 private:
     void _build_slices() const;
 
+    void _check_overvlow() const {
+        if constexpr (std::is_same_v<T, uint32_t>) {
+            if (UNLIKELY(_bytes.size() > std::numeric_limits<uint32_t>::max())) {
```
Comment (stdpain) [inline be/src/column/binary_column.h:348]: we shouldn't throw exception any where. some scope was not in catched scope

Comment (liuyehcf) [reply be/src/column/binary_column.h]: updated

**Review by silverbullet233 — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**